src/hooks/useScrollZoom.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+import { useUiStateStore } from 'src/stores/uiStateStore';
+
+
+const useScrollZoom = () => {
+  const uiStateActions = useUiStateStore((state) => {
+    return state.actions;
+  });
+  const disableScrollZoom = useUiStateStore((state) => {
+    return state.disableScrollZoom;
+  });
+
+  const handleScroll = (e: WheelEvent) => {
+    if (disableScrollZoom) return;
+    if (e.deltaY > 0) {
+        uiStateActions.decrementZoom();
+      } else {
+        uiStateActions.incrementZoom();
+      }
+  };
+
+  useEffect(() => {
+    window.addEventListener('wheel', handleScroll);
```

Comment:
Following on from my first comment about preventing simultaneous zooming and scrolling, if you target `uiStateStore.rendererEl` instead of `window`, this should listen out only to interactions on the editor canvas (and avoid interactions on overlay elements like the toolbar, sidepanels and burger menu).

This should make things simpler.